app/src/test/java/ch/epfl/sweng/erpa/services/GCP/GCPDataServiceTest.java

```diff
@@ -0,0 +1,71 @@
+package ch.epfl.sweng.erpa.services.GCP;
+
+import com.annimon.stream.Optional;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+
+import java.io.IOException;
+
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public abstract class GCPDataServiceTest<T> {
+    protected final Retrofit retrofit = new Retrofit.Builder()
+            .baseUrl(GCPRemoteServicesProvider.API_URL)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build();
+    protected T obj;
+
+    protected abstract String prefix();
+    public GCPRemoteServicesProvider getGrsp() {
+        return grsp;
+    }
+
+    public ArgumentCaptor<Call<T>> getArgumentCaptor0() {
+        return argumentCaptor0;
+    }
+
+    @InjectMocks private GCPRemoteServicesProvider grsp;
+    @Captor private ArgumentCaptor<Call<T>> argumentCaptor0;
+
+
+    @Before
+    public void setUp() throws IOException, ServerException {
+        grsp = mock(GCPRemoteServicesProvider.class);
+        when(grsp.getRetrofit()).thenReturn(retrofit);
+        when(grsp.callAndReturnOptional(argumentCaptor0.capture())).thenReturn(Optional.of(obj));
+    }
+
+    @Test
+    public void testUpdate() throws IOException, ServerException {
+        verify(getGrsp(), times(1)).executeAndThrowOn404(getArgumentCaptor0().capture());
```

Comment:
I don't understand these tests. I can't see the different steps
- Given
- When
- Then

Reply:
Me neither